components/eamxx/src/physics/mam/eamxx_mam_constituent_fluxes_interface.cpp

```diff
@@ -0,0 +1,291 @@
+      "Error! Used memory != requested memory for MAMConstituentFluxes.");
+}
+
+// ================================================================
+//  INITIALIZE_IMPL
+// ================================================================
+void MAMConstituentFluxes::initialize_impl(const RunType run_type) {
+  // ---------------------------------------------------------------
+  // Input fields read in from IC file, namelist or other processes
+  // ---------------------------------------------------------------
+
+  // Populate the wet atmosphere state with views from fields
+  // FIMXE: specifically look which among these are actually used by the process
+  wet_atm_.qv = get_field_in("qv").get_view<const Real **>();
+
+  // Following wet_atm vars are required only for building DS
+  wet_atm_.qc = get_field_in("qc").get_view<const Real **>();
+  wet_atm_.nc = get_field_in("nc").get_view<const Real **>();
+  wet_atm_.qi = get_field_in("qi").get_view<const Real **>();
+  wet_atm_.ni = get_field_in("ni").get_view<const Real **>();
+
+  // Populate the dry atmosphere state with views from fields
+  dry_atm_.T_mid = get_field_in("T_mid").get_view<const Real **>();
+  dry_atm_.p_mid = get_field_in("p_mid").get_view<const Real **>();
+  dry_atm_.p_del = get_field_in("pseudo_density").get_view<const Real **>();
+  dry_atm_.p_int = get_field_in("p_int").get_view<const Real **>();
+
+  // Following dry_atm vars are required only for building DS
+  dry_atm_.cldfrac = get_field_in("cldfrac_tot").get_view<const Real **>();
+  dry_atm_.pblh    = get_field_in("pbl_height").get_view<const Real *>();
+  dry_atm_.omega   = get_field_in("omega").get_view<const Real **>();
+  dry_atm_.p_del   = get_field_in("pseudo_density").get_view<const Real **>();
+
+  // store fields converted to dry mmr from wet mmr in dry_atm_
+  dry_atm_.z_mid     = buffer_.z_mid;
+  dry_atm_.z_iface   = buffer_.z_iface;
+  dry_atm_.dz        = buffer_.dz;
+  dry_atm_.qv        = buffer_.qv_dry;
+  dry_atm_.qc        = buffer_.qc_dry;
+  dry_atm_.nc        = buffer_.nc_dry;
+  dry_atm_.qi        = buffer_.qi_dry;
+  dry_atm_.ni        = buffer_.ni_dry;
+  dry_atm_.w_updraft = buffer_.w_updraft;
+  dry_atm_.z_surf    = 0.0;  // FIXME: for now
+
+  // Constituent fluxes at the surface (gasses and aerosols) [kg/m2/s]
+  constituent_fluxes_ =
+      get_field_in("constituent_fluxes").get_view<const Real **>();
+
+  // interstitial and cloudborne aerosol tracers of interest: mass (q) and
+  // number (n) mixing ratios
+  for(int m = 0; m < mam_coupling::num_aero_modes(); ++m) {
+    // interstitial aerosol tracers of interest: number (n) mixing ratios
+    const char *int_nmr_field_name = mam_coupling::int_aero_nmr_field_name(m);
+    wet_aero_.int_aero_nmr[m] =
+        get_field_out(int_nmr_field_name).get_view<Real **>();
+
+    // cloudborne aerosol tracers of interest: number (n) mixing ratios
+    const char *cld_nmr_field_name = mam_coupling::cld_aero_nmr_field_name(m);
```

Comment:
Same comment as above.

Reply:
Yes, we will clean this up to make it much simpler.

Reply:
Not in this PR but in our clean-up round.